debugger/debugrt/host/host.c

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/debugrt/host.h>
+#include <string.h>
+
+/**
+ * In Windows, debugrt is built as a separate DLL that
+ * OE host applications call into. Hence, this module cannot
+ * use functionality (e.g spinlocks) defined in oehost.
+ */
+
+#ifdef _MSC_VER
+
+#include <Windows.h>
+
+static uint64_t _lock = 0;
+
+static void spin_lock()
+{
+    while (InterlockedCompareExchange(&_lock, 1, 0) == 1)
+    {
+        // TODO: Do we need to yield CPU here?
```

Comment:
In reference to: https://github.com/openenclave/openenclave/pull/2020#discussion_r299258976

I was not suggesting that this needs to take a dependency on oehost library (I don't believe we have an `oe_spinlock` implementation on the host side, do we?)

I'm suggesting that you consider using an appropriate OS implementation of the appropriate constructs instead of reimplementing using atomics; most implementations will already take into account falling back to a wait if spinning for too long instead of us reinventing the wheel here. This code already forks on windows/linux so that doesn't seem to be a consideration.

Also consider if spinlocks are appropriate here if we don't have to implement from scratch, or something more generic like futex.

Reply:
Ah ok. Good point. Have added it to #2020.